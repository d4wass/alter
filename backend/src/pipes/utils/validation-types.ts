import { ValidationError } from '@nestjs/common';

export type ConvertedValidationError = {
  property: string;
  constraints: any;
  children?: ValidationError[];
};

export type ConstructorTypes =
  | StringConstructor
  | BooleanConstructor
  | NumberConstructor
  | ArrayConstructor
  | ObjectConstructor;
