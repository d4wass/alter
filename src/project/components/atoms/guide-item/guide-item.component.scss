@use '../../../../mixins';
@use '../../../../base';

.img-container{
    display: flex;
    justify-content: center;
    align-items: center;
    img{
        width: 200px;
        height: 200px;
    }
}

.wrapper{
    @include mixins.subtitle(25px);
    @include mixins.paragraph;
    border-radius: base.$main-border-rad;
    box-shadow: 0 4px 6px 0 hsla(0, 0%, 0%, 0.2);
    height: 100%;
    padding: 20px;

    h3{
        margin: 0
    }
}