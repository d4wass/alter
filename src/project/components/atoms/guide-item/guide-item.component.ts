import { Component, Input } from '@angular/core';

@Component({
    selector: 'app-guide-item',
    template: `
    <div class="wrapper">
      <div class="img-container">
        <img src="{{ graphic }}" *ngIf="graphic" data-test="image" />
      </div>
      <h3 data-test="title">{{ title }}</h3>
      <p data-test="content">{{ content }}</p>
    </div>
  `,
    styleUrls: ['./guide-item.component.scss'],
    standalone: false
})
export class GuideItemComponent {
  @Input() title!: string;
  @Input() content!: string;
  @Input() graphic?: string;
}
