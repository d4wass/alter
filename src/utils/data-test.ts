export function dataTest(selector: string) {
  return `[data-test="${selector}"]`;
}
