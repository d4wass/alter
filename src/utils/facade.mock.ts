import { Observable, BehaviorSubject } from 'rxjs';

export type FacadeMock<T> = {
  [P in keyof T]: T[P] extends Observable<infer U>
    ? BehaviorSubject<U>
    : T[P] extends (args: infer A) => Observable<infer R>
    ? (args: A) => BehaviorSubject<R>
    : T[P];
};
