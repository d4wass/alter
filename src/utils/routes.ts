export interface urlItem {
  description: string;
  url: string;
}

export const urlList: urlItem[] = [
  { description: 'Learn more', url: 'faq' },
  { description: 'Carculator', url: 'carculator' },
  { description: 'Become a host', url: 'host' },
  { description: 'Create Account', url: 'new-user' },
  { description: 'Your Cars', url: 'vehicles' }
];
