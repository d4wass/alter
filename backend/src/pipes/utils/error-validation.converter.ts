import { ValidationError } from 'class-validator';
import { ConvertedValidationError } from './validation-types';

export class ValidationErrorConverter {
  private _convertedErrors: Array<ConvertedValidationError> = [];
  public get convertedErrors(): Array<ConvertedValidationError> {
    return this._convertedErrors;
  }
  public set convertedErrors(value: Array<ConvertedValidationError>) {
    this._convertedErrors = value;
  }
  private _errors: ValidationError[];

  constructor(errors: ValidationError[]) {
    this._errors = errors;
    this.errorsConverter(this._errors);
  }

  errorsConverter(errors: ValidationError[]): void {
    errors.forEach((error) => {
      if (!error.children.length) {
        this._convertedErrors.push({ property: error.property, constraints: error.constraints });
      } else {
        this.errorsConverter(error.children);
      }
    });
  }
}
