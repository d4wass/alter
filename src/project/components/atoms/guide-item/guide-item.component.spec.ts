import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { dataTest } from '../../../../utils/data-test';
import { GuideItemComponent } from './guide-item.component';

describe('GuideItemComponent', () => {
  let spectator: Spectator<GuideItemComponent>;
  let component: GuideItemComponent;

  const createComponent = createComponentFactory(GuideItemComponent);
  const selectors = {
    title: () => spectator.query(dataTest('title')),
    content: () => spectator.query(dataTest('content')),
    image: () => spectator.query(dataTest('image'))
  };

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
  //TODO: check if elements of component recived proper value from inputs
  //TODO: check if <img> is displayed when @graphic when was setted and conversly

  describe('@Inputs', () => {
    describe('set input by value', () => {
      beforeEach(() => {
        component.title = 'Title';
        component.content = 'content';
        component.graphic = 'graphic';

        spectator.detectComponentChanges();
      });

      it('should [title] setted with given value', () => {
        expect(component.title).toEqual('Title');
      });
      it('should [content] setted with given value', () => {
        expect(component.content).toEqual('content');
      });
      it('should [graphic] setted with given value', () => {
        expect(component.graphic).toEqual('graphic');
      });
    });

    it('should set [graphic] to indefined when input not received any value', () => {
      expect(component.graphic).toBeUndefined();
    });
  });

  describe('Template', () => {});
});
