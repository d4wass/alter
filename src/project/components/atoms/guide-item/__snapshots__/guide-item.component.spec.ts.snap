// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideItemComponent should match to snapshot 1`] = `
<app-guide-item>
  <div
    class="wrapper"
  >
    <div
      class="img-container"
    >
    </div>
    <h3
      data-test="title"
    >
    </h3>
    <p
      data-test="content"
    >
    </p>
  </div>
</app-guide-item>
`;
